source/common/upstream/upstream_impl.cc

```diff
@@ -1952,12 +1994,68 @@ ClusterInfoImpl::ResourceManagers::load(const envoy::config::cluster::v3::Cluste
       max_connections_per_host = per_host_it->max_connections().value();
     }
   }
-  return std::make_unique<ResourceManagerImpl>(
-      runtime, runtime_prefix, max_connections, max_pending_requests, max_requests, max_retries,
-      max_connection_pools, max_connections_per_host,
-      ClusterInfoImpl::generateCircuitBreakersStats(stats_scope, priority_stat_name,
-                                                    track_remaining, circuit_breakers_stat_names_),
-      budget_percent, min_retry_concurrency);
+  return std::make_unique<ResourceManagerImpl>(runtime, runtime_key_prefix, max_connections,
+                                               max_pending_requests, max_requests, max_retries,
+                                               max_connection_pools, max_connections_per_host,
+                                               cb_stats, budget_percent, min_retry_concurrency);
+}
+
+AdmissionControlImplSharedPtr ClusterInfoImpl::CircuitBreakers::loadAdmissionControl(
+    const envoy::config::cluster::v3::Cluster& config,
+    const envoy::config::core::v3::RoutingPriority& priority, ClusterCircuitBreakersStats cb_stats,
+    ProtobufMessage::ValidationVisitor& validation_visitor, Runtime::Loader& runtime,
+    const std::string& runtime_key_prefix) {
+  const auto& thresholds = config.circuit_breakers().thresholds();
+  const auto it = std::find_if(
+      thresholds.cbegin(), thresholds.cend(),
+      [priority](const envoy::config::cluster::v3::CircuitBreakers::Thresholds& threshold) {
+        return threshold.priority() == priority;
+      });
+
+  absl::optional<envoy::config::cluster::v3::CircuitBreakers::Thresholds> pri_thresholds =
+      absl::nullopt;
+  if (it != thresholds.cend()) {
+    pri_thresholds = *it;
+  }
+  envoy::config::core::v3::TypedExtensionConfig retry_admission_control =
+      ClusterInfoImpl::getRetryAdmissionControlConfig(pri_thresholds);
+
+  return std::make_shared<AdmissionControlImpl>(retry_admission_control, validation_visitor,
+                                                runtime, runtime_key_prefix, cb_stats);
+}
+
+envoy::config::core::v3::TypedExtensionConfig ClusterInfoImpl::getRetryAdmissionControlConfig(
+    const absl::optional<envoy::config::cluster::v3::CircuitBreakers::Thresholds>& thresholds) {
```

Comment:
We are deprecating the ResourceManager version of retry circuit breaking by dynamically translating the old config into extension config. This has the upside of moving people off the old config quicker, allowing us to remove the old code before the fields are fully removed.

However, it does mean that given the circuit breaking defaults, the two new extensions are required.